backend/src/zimfarm_backend/common/schemas/offliners/devdocs.py

```diff
@@ -1,151 +1,110 @@
-from marshmallow import fields
-
-from common.schemas import SerializableSchema, String
-from common.schemas.fields import (
-    validate_output,
-    validate_zim_description,
-    validate_zim_longdescription,
-    validate_zim_title,
+from zimfarm_backend.common.schemas import BaseModel
+from zimfarm_backend.common.schemas.fields import (
+    OptionalField,
+    OptionalNotEmptyString,
+    OptionalZIMDescription,
+    OptionalZIMLongDescription,
+    OptionalZIMOutputFolder,
+    OptionalZIMTitle,
 )
 
 
-class DevDocsFlagsSchema(SerializableSchema):
-    class Meta:
-        ordered = True
-
-    slug = String(
-        metadata={
-            "label": "Slug",
-            "description": "Fetch the provided Devdocs resource. "
-            "Slugs are the first path entry in the Devdocs URL. "
-            "For example, the slug for: `https://devdocs.io/gcc~12/` is `gcc~12`. "
-            "Mutually exclusive with `All` setting, set only one option. Either this"
-            "setting or `All` must be configured.",
-        },
+class DevDocsFlagsSchema(BaseModel):
+    slug: OptionalNotEmptyString = OptionalField(
+        title="Slug",
+        description="""Fetch the provided Devdocs resource.
+            Slugs are the first path entry in the Devdocs URL.
+            For example, the slug for: `https://devdocs.io/gcc~12/` is `gcc~12`.
+            Mutually exclusive with `All` setting, set only one option. Either this
+            setting or `All` must be configured.""",
     )
 
-    all_flag = fields.Boolean(
-        truthy=[True],
-        falsy=[False],
-        metadata={
-            "label": "All",
-            "description": "Fetch all Devdocs resources, and produce one ZIM "
-            "per resource. Mutually exclusive with `Slug` setting, set only "
-            "one option. Either this setting or `Slug` must be configured.",
-        },
-        data_key="all",
+    all_flag: bool | None = OptionalField(
```

Comment:
If I add the generator which converts snake_case to kebab-case, how would it be used for this class?